shoes-core/lib/shoes/line.rb

```diff
@@ -19,6 +21,26 @@ def create_dimensions(point_a, point_b)
       style[:y2] = point_b.y
     end
 
+    # Check out http://math.stackexchange.com/questions/60070/checking-whether-a-point-lies-on-a-wide-line-segment
+    # for explanations how the algorithm works
+    def in_bounds?(x, y)
+      # c is (x, y)
+      left_most, right_most = point_a.x < point_b.x ? [point_a, point_b] : [point_b, point_a]
+      vec_left_c = Vector[left_most.x - x, left_most.y - y]
```

Comment:
`vec_left_c = Vector[left_most.x - x, left_most.y - y]`
Maybe it'd be better to add `+/-` methods to `Shoes::Point`?